packages/driver/src/cy/commands/navigation.coffee

```diff
@@ -564,6 +564,11 @@ module.exports = (Commands, Cypress, cy, state, config) ->
         ## for this, and so we need to resolve onLoad immediately
         ## and bypass the actual visit resolution stuff
         if bothUrlsMatchAndRemoteHasHash(current, remote)
+          if current.hash is remote.hash
```

Comment:
Also good idea to add github URL to the issue this is fixing to help explain why this logic is necessary.